syntax = "proto3";
package message;

message MessageObjectResponse {
  int64 message_id = 1;
  string group_id = 2;
  string from_client_id = 3;
  string client_id = 4;
  bytes message = 5;
  int64 created_at = 6;
  int64 updated_at = 7;
}

message BaseResponse {
    bool success = 1;
}

// Request: get list message group
message GetMessagesInGroupRequest {
  string group_id = 1;
  int32 off_set = 2;
  int64 last_message_at = 3;
}
// Response: GetMessagesInGroupResponse
message GetMessagesInGroupResponse {
  repeated MessageObjectResponse lst_message = 1;
}


// Method
service Message {
    rpc get_messages_in_group(GetMessagesInGroupRequest) returns (GetMessagesInGroupResponse) {};
}
